### PR TITLE
Upgrade amplify to v6

### PR DESCRIPTION
**Pull request by gmrabian**
### Description

This change upgrades Amplify from v5 to v6 

The upgrade work was previously done in [CARTS](https://github.com/Enterprise-CMCS/macpro-mdct-carts/pull/139616) but was reverted due to an [issue](https://github.com/aws-amplify/amplify-js/issues/13310) that has since been resolved by the maintainers.

References:
[upgrade guide](https://docs.amplify.aws/gen1/react/build-a-backend/troubleshooting/migrate-from-javascript-v5-to-v6/)

### Related ticket(s)

CMDCT-4035

---
### How to test

Everything in the app works as expected. A short list of everything:
- all api calls
- auth
- timeout
- template file download

### Notes

---
### Pre-review checklist

- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- ~[ ] I have updated relevant documentation, if necessary~
- [x] I have performed a self-review of my code
- [x] I have manually tested this PR in the deployed cloud environment

### CODE REVIEW

**Review by karla-vm — APPROVED** [review]

File: services/ui-src/src/components/logins/LoginCognito.test.tsx [hunk 1]
```diff
@@ -12,20 +24,19 @@ const loginCognitoComponent = (
...
+    const passwordInput = screen.getByLabelText("Password");
+    const submitButton = screen.getByRole("button");
+    await userEvent.type(emailInput, "email@address.com");
+    await userEvent.type(passwordInput, "p@$$w0rd"); //pragma: allowlist secret
+    await userEvent.click(submitButton);
+    await expect(mockSignIn).toHaveBeenCalledWith({
+      username: "email@address.com",
+      password: "p@$$w0rd", //pragma: allowlist secret
+    });
+    await expect(mockUseNavigate).toHaveBeenCalledWith("/");
```
Comment (JonHolman) [inline services/ui-src/src/components/logins/LoginCognito.test.tsx:39]: Do we need to await these expects?

File: services/ui-src/src/utils/api/requestMethods/getTemplateUrl.ts [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
   };
 
-  const response = await API.get("mcr", `/templates/${templateName}`, request);
-  return response;
+  const { body } = await get({
+    apiName: "mcr",
+    path: `/templates/${templateName}`,
+    options,
+  }).response;
+  return (await body.json()) as string;
```
Comment (JonHolman) [inline services/ui-src/src/utils/api/requestMethods/getTemplateUrl.ts:15]: does body.json() already return a string?

**Review by JonHolman — DISMISSED** [review]
Minor questions.  Approved.

File: services/ui-src/src/components/reports/ReportProvider.tsx [hunk 3]
```diff
@@ -79,7 +79,7 @@ export const ReportProvider = ({ children }: Props) => {
       hydrateAndSetReport(result);
       return result;
     } catch {
-      setError(reportErrors.GET_REPORT_FAILED);
+      return setError(reportErrors.GET_REPORT_FAILED);
```
Comment (bangbay-bluetiger) [inline services/ui-src/src/components/reports/ReportProvider.tsx:82]: Do we really want to return this method? Or do this instead?
```
setError(reportErrors.GET_REPORT_FAILED);
return;
```

File: services/ui-src/src/utils/api/requestMethods/banner.ts [hunk 4]
```diff
@@ -1,40 +1,55 @@
...
 
   updateTimeout();
-  const response = await API.get("mcr", `/banners/${bannerKey}`, request);
-  return response;
+  const { body } = await get({
+    apiName,
+    path,
+    options,
+  }).response;
+  return (await body.json()) as unknown as { Item: AdminBannerData };
```
Comment (bangbay-bluetiger) [inline services/ui-src/src/utils/api/requestMethods/banner.ts:21]: Is casting to `unknown` necessary?

File: services/ui-src/src/utils/api/requestMethods/report.ts [hunk 5]
```diff
@@ -1,85 +1,90 @@
...
-    `/reports/submit/${reportType}/${state}/${id}`,
-    request
-  );
-  return response;
+  const { body } = await post({
+    apiName,
+    path,
+    options,
+  }).response;
+  return (await body.json()) as unknown as ReportShape;
```
Comment (bangbay-bluetiger) [inline services/ui-src/src/utils/api/requestMethods/report.ts:54]: Same question about `unknown`. I'm guessing that the casting makes more sense where these endpoints are actually called? But within these methods it isn't immediately obvious why the casting needs to happen.

File: services/ui-src/src/utils/auth/UserProvider.tsx [hunk 6]
```diff
@@ -65,9 +65,15 @@ export const UserProvider = ({ children }: Props) => {
     }
 
     try {
-      const session = await Auth.currentSession();
-      const payload = session.getIdToken().payload;
-      const { email, given_name, family_name } = payload;
+      const tokens = (await fetchAuthSession()).tokens;
+      if (!tokens || !tokens.idToken) {
```
Comment (bangbay-bluetiger) [inline services/ui-src/src/utils/auth/UserProvider.tsx:69]: `!tokens?.idToken`?

**Review by bangbay-bluetiger — COMMENTED** [review]
Looks like there's e2e tests failing. Don't know if they're related to changes from the PR.

File: services/ui-src/src/components/logins/LoginCognito.test.tsx (hunk 1 above)
Comment (gmrabian) [reply services/ui-src/src/components/logins/LoginCognito.test.tsx:39]: we do not! good catch.

File: services/ui-src/src/utils/api/requestMethods/banner.ts (hunk 4 above)
Comment (gmrabian) [reply services/ui-src/src/utils/api/requestMethods/banner.ts:21]: the [build was failing](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/actions/runs/11280305608/job/31373091951) previously so Ben pushed [this change](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/pull/11876/commits/5ab08c6b4b9c9f0eb3d6e321cf6d6e3177a6636e) to resolve it. I too don’t love the casting but I need the build to pass. Open to alternatives!

File: services/ui-src/src/utils/api/requestMethods/report.ts (hunk 5 above)
Comment (gmrabian) [reply services/ui-src/src/utils/api/requestMethods/report.ts:54]: the [build was failing](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/actions/runs/11280305608/job/31373091951) previously so Ben pushed [this change](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/pull/11876/commits/5ab08c6b4b9c9f0eb3d6e321cf6d6e3177a6636e) to resolve it. I too don’t love the casting but I need the build to pass. Open to alternatives!

File: services/ui-src/src/utils/api/requestMethods/getTemplateUrl.ts (hunk 2 above)
Comment (JonHolman) [reply services/ui-src/src/utils/api/requestMethods/getTemplateUrl.ts:15]: Oh, interesting!  Yeah, it looks like that's necessary.  The only alternative I think of is: 
`String(await body.json())` that's just style preference.

**Review by bangbay-bluetiger — APPROVED** [review]

**Review by jessabean — APPROVED** [review]

**Pull request merged**